dialog_lib/agents/abstract.py

```diff
@@ -141,29 +142,57 @@ def documents_formatter(self, docs, document_separator="\n\n"):
         return document_separator.join(doc_strings)
 
     @property
-    def context_dict(self):
+    def retriever_chain(self):
         """
-        builds and returns the context dictionary for the LCEL
+        builds and returns the retriever chain for the LCEL
         """
+        return (
+            itemgetter("input") | self.retriever
+        ).with_config({"run_name": "RetrieverChain"})
 
-        context_dict = {
-            "input": RunnablePassthrough(),
-            "chat_history": itemgetter("chat_history"),
-        }
+    @property
+    def fallback_chain(self):
+        """
+        builds and returns the fallback message chain for the LCEL
+        """
+        fallback_prompt = ChatPromptTemplate.from_messages(
+            [
+                ("ai", self.config.get("prompt").get("fallback_not_found_relevant_contents"))
+            ]
+        )
 
-        if self.retriever:
-            context_dict["context"] = itemgetter("input") | self.retriever | self.documents_formatter
+        def parse_fallback(ai_message):
+            return ai_message.content
```

Comment:
is this used? I think it'd be used after 
```
fallback_prompt | RunnableLambda(lambda x: x.messages[-1])
```
to parse the ai message content out to the user.

Reply:
Totally not used. We are solving this in the post-processing function, so no need for it!